phoenix-core/src/main/java/org/apache/phoenix/mapreduce/transform/TransformTool.java

```diff
@@ -0,0 +1,761 @@
+        }
+
+        if (useEndTime) {
+            endTime = new Long(cmdLine.getOptionValue(END_TIME_OPTION.getOpt()));
+        }
+
+        if (isTimeRangeSet(startTime, endTime)) {
+            validateTimeRange(startTime, endTime);
+        }
+
+        if (isPartialTransform &&
+                (cmdLine.hasOption(AUTO_SPLIT_OPTION.getOpt()))) {
+            throw new IllegalArgumentException(PARTIAL_TRANSFORM_NOT_COMPATIBLE);
+        }
+        if (isPartialTransform &&
+                (cmdLine.hasOption(ABORT_TRANSFORM_OPTION.getOpt()) || cmdLine.hasOption(PAUSE_TRANSFORM_OPTION.getOpt())
+                        || cmdLine.hasOption(RESUME_TRANSFORM_OPTION.getOpt()))) {
+            throw new IllegalArgumentException(PARTIAL_TRANSFORM_NOT_COMPATIBLE);
+        }
+
+        if (isPartialTransform) {
+            if (!cmdLine.hasOption(START_TIME_OPTION.getOpt())) {
+                throw new IllegalArgumentException(PARTIAL_TRANSFORM_NOT_APPLICABLE);
+            }
+            lastTransformTime = new Long(cmdLine.getOptionValue(START_TIME_OPTION.getOpt()));
+            SystemTransformRecord transformRecord = getTransformRecord(null);
+            if (transformRecord == null) {
+                throw new IllegalArgumentException(PARTIAL_TRANSFORM_NOT_APPLICABLE);
+            }
+            if (lastTransformTime == null) {
+                lastTransformTime = transformRecord.getTransformEndTs().getTime();
+            } else {
+                validateLastTransformTime();
+            }
+        }
+
+        schemaName = cmdLine.getOptionValue(SCHEMA_NAME_OPTION.getOpt());
+        dataTable = cmdLine.getOptionValue(DATA_TABLE_OPTION.getOpt());
+        indexTable = cmdLine.getOptionValue(INDEX_TABLE_OPTION.getOpt());
+        qDataTable = SchemaUtil.getQualifiedTableName(schemaName, dataTable);
+        isForeground = cmdLine.hasOption(RUN_FOREGROUND_OPTION.getOpt());
+        logicalTableName = dataTable;
+        logicalParentName = null;
+        if (!Strings.isNullOrEmpty(indexTable)) {
+            logicalTableName = indexTable;
+            logicalParentName = SchemaUtil.getTableName(schemaName, dataTable);
+        }
+
+        pDataTable = PhoenixRuntime.getTable(
+                connection, SchemaUtil.getQualifiedTableName(schemaName, dataTable));
+        if (indexTable != null) {
+            pIndexTable = PhoenixRuntime.getTable(
+                    connection, SchemaUtil.getQualifiedTableName(schemaName, indexTable));
+            pOldTable = pIndexTable;
+        } else {
+            pOldTable = pDataTable;
+        }
+
+        SystemTransformRecord transformRecord = getTransformRecord(connection.unwrap(PhoenixConnection.class));
+
+        validateTransform(pDataTable, pIndexTable, transformRecord);
+        String newTableName = SchemaUtil.getTableNameFromFullName(transformRecord.getNewPhysicalTableName());
+        pNewTable = PhoenixRuntime.getTable(
```

Comment:
should this be getTableNoCache? Does it matter if our copy of PTable is slightly stale?